doc/extending.md

```diff
@@ -61,3 +61,84 @@ can run the templating manually:
 ```
 ./baselayer/tools/fill_conf_values.py --config="config.yaml" static/js/component.jsx.template
 ```
+
+## Adding external services
+
+External services are microservices that you can pull from GitHub and run as part of your application. This is useful for integrating third-party services or custom scripts.
+
+If you want to add external services to your application,
+you can do so by adding them to the `config.yaml` file under the `services.external` key. This allows baselayer to pull the external service from a GitHub repository.
+
+The configuration in the `config.yaml` file should look like this:
+
+```
+services:
+  external:
+    my_service:
+        url: "https://github.com/my_service.git"
+        branch: main
+        sha: specific_commit_sha
+    another_service:
+        url: "https://github.com/another_service.git"
+        version: v0.1.0
+        params:
+            endpoint: "https://api.example.com"
+```
+
+You must provide the `url` of the GitHub repository, as well as a version of the service. To specify the version, you are required to either:
+
+- Provide both the `branch` and `sha`,
+  **or**
+- Use the `version` field to reference a specific release.
+
+Additional configuration options can be passed through the `params` dictionary, which the service uses.
+
+The external service will then be initialized and registered in `supervisor` alongside other services, provided it is correctly configured and compatible with the application.
+Compatibility can be enforced via version constraints declared in the service’s optional pyproject.toml. See [External Service Requirements](#external-service-requirements) for details.
+
+### External Service Requirements
+
+To work correctly with the application, external services should follow these conventions:
+
+#### Entry Point
+
+- The service should include a `main.py` file as its entry point. In that case, providing a `supervisor.conf` can be omitted and will be auto-generated, referring to `main.py`.
+- If the entry point differs, a `supervisord.conf` must be provided in the repository, pointing to the correct entry point.
+
+#### Project Metadata and Compatibility
+
+External services may include a `pyproject.toml` file to provide metadata and compatibility information. This is especially important when multiple applications (or versions) are built on top of Baselayer.
```

Comment:
What does this mean: "especially important when multiple applications (or versions) are built on top of Baselayer"?